test/base_test.py

```diff
@@ -81,14 +81,11 @@ def __init__(self, shape, typechar, itemsize):
 
     def assertSame(self, proxy, obj):
         self.assertEqual(proxy.length, obj.size)
-        d = proxy.__array_interface__
-        try:
-            self.assertEqual(d['typestr'], obj.typestr)
-            self.assertEqual(d['shape'], obj.shape)
-            self.assertEqual(d['strides'], obj.strides)
-            self.assertEqual(d['data'], obj.data)
-        finally:
-            d = None
+        iface = proxy.__array_interface__
```

Comment:
I wonder why you moved this out of a try finally block?

Probably the reason it assigns that to None is so that the destructor is called immediately.

Reply:
ok, I just wondered about all these `del sth`, I might have been going too far... 
But why should the destructor be called immediately? After all, the tests pass without...

Reply:
I guess this will cause issues on pypy, which doesn't call destructors when something leaves scope.